helm-chart/binderhub/templates/deployment.yaml

```
@@ -102,8 +102,12 @@ spec:
           value: {{ .Values.jupyterhub.hub.services.binder.oauth_client_id | quote}}
         - name: JUPYTERHUB_OAUTH_CALLBACK_URL
           value: {{ .Values.jupyterhub.hub.services.binder.oauth_redirect_uri | quote}}
+        {{- if .Values.jupyterhub.hub.allowNamedServers }}
         - name: JUPYTERHUB_ALLOW_NAMED_SERVERS
-          value: {{ .Values.jupyterhub.hub.allowNamedServers | quote }}
+          value: true
+        - name: JUPYTERHUB_NAMED_SERVER_LIMIT_PER_USER
+          value: {{ .Values.jupyterhub.hub.namedServerLimitPerUser }}
```

```suggestion
          value: {{ .Values.jupyterhub.hub.namedServerLimitPerUser | quote  }}
```
As before this needs to be quoted other I get an error. However after adding `| quote` I run into an error in BinderHub:
```
traitlets.traitlets.TraitError: The 'named_server_limit_per_user' trait of a Launcher instance must be an int, but a value of '2' <class 'str'> was specified.
```

---

This is weird :-/ Are you sure you tried this with the old way of specifying the value? My guess would be that it didn't use to work previously either. `os.getenv("JUPYTERHUB_NAMED_SERVER_LIMIT_PER_USER", 0)` should always return a string (if the variable is set). So we should always have had this problem.

Maybe the code to fetch the variable should be `int(os.getenv("JUPYTERHUB_NAMED_SERVER_LIMIT_PER_USER", "0"))` instead. Which makes me wonder: can't the traitlet take care of fetching values from the env for us and dealing with the conversion? (I suspect not)

---

I might've been unclear, "as before" refers to my [earlier comment in this PR on quoting](https://github.com/jupyterhub/binderhub/pull/1029#discussion_r372679000). `JUPYTERHUB_NAMED_SERVER_LIMIT_PER_USER` is new in this PR.

I agree it would be nice if traitlets could optionally cast values, but they don't.